OpenRA.Platforms.Default/OpenAlSoundEngine.cs

```diff
@@ -56,14 +56,17 @@ static string[] QueryDevices(string label, int type)
 
 			var devices = new List<string>();
 			var next = ALC10.alcGetString(IntPtr.Zero, type);
-			do
+			if (next != IntPtr.Zero)
 			{
-				var str = Marshal.PtrToStringAnsi(next);
-				next += str.Length + 1;
-				devices.Add(str);
-			} while (Marshal.ReadByte(next) != 0);
+				do
+				{
+					var str = Marshal.PtrToStringAnsi(next);
+					next += str.Length + 1;
+					devices.Add(str);
+				} while (Marshal.ReadByte(next) != 0);
+			}
 
-			if (AL10.alGetError() != AL10.AL_NO_ERROR)
+			if (next == IntPtr.Zero || AL10.alGetError() != AL10.AL_NO_ERROR)
```

Comment:
It looks like you could move this above the dowhile and avoid the new IntPtr.Zero check.